token-client/src/main/java/com/sap/cloud/security/xsuaa/client/SpringOAuth2TokenKeyService.java

```diff
@@ -37,20 +36,27 @@ public SpringOAuth2TokenKeyService(@Nonnull RestOperations restOperations) {
 	@Override
 	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId)
 			throws OAuth2ServiceException {
-		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, null);
+		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, null, null);
 	}
 
 	@Override
 	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId)
 			throws OAuth2ServiceException {
+		return retrieveTokenKeys(tokenKeysEndpointUri, tenantId, clientId, null);
+	}
+
+	@Override
+	public String retrieveTokenKeys(@Nonnull URI tokenKeysEndpointUri, @Nullable String tenantId, @Nullable String clientId, @Nullable String azp)
```

Comment:
refactor this method as well, same as above in the OAuth2TokenKeyServiceWithCache.java